state/application.go

```diff
@@ -709,7 +710,17 @@ func (a *Application) changeCharmOps(
 ) ([]txn.Op, error) {
 	// Build the new application config from what can be used of the old one.
 	var newSettings charm.Settings
-	oldKey, err := readSettings(a.st.db(), settingsC, a.charmConfigKey())
+
```

Comment:
Changes to the logic accessing the "charms" collection also need to be instituted.